### PR TITLE
CPLP-751 - adjust entities to match devenv PR#8

### PR DESCRIPTION
**Pull request by ntruchsess**
Adjustment of EF-Core entities to match https://github.com/tractusx-team-portal-onboarding/devenv/pull/8

! currently not ready to merge as there are issues with nullable reference-typed entity members. !

### CODE REVIEW

**Comment by ntruchsess** [comment]
snake-case maps 'Alpha2Code' to 'alpha2code', not 'alpha_2_code' so I renamed the column in the database: https://github.com/tractusx-team-portal-onboarding/devenv/pull/16

**Review by ntruchsess — COMMENTED** [review]
@wira-mms : I Did check your changes - they look fine.

**Review by wira-mms — CHANGES_REQUESTED** [review]

**Review by ntruchsess — COMMENTED** [review]
Datetime-relevant changes are fine and approved

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs [hunk 1]
```diff
@@ -56,81 +56,86 @@ public IAsyncEnumerable<string> GetIdpAliaseForCompanyIdUntrackedAsync(Guid comp
...
         public IdentityProvider CreateSharedIdentityProvider(Company company)
         {
-            var idp = new IdentityProvider() {
-                Id = Guid.NewGuid(),
-                IdentityProviderCategoryId = IdentityProviderCategoryId.KEYCLOAK_SHARED,
-            };
+            var idp = new IdentityProvider(
+                Guid.NewGuid(),
+                IdentityProviderCategoryId.KEYCLOAK_SHARED,
+                DateTime.Now);
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs]: implementd

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/IamIdentityProvider.cs [hunk 2]
```diff
@@ -5,17 +5,24 @@ namespace CatenaX.NetworkServices.PortalBackend.PortalEntities.Entities
 {
     public class IamIdentityProvider
     {
-        public IamIdentityProvider() {}
-        public IamIdentityProvider(string iamIdpAlias)
+        public IamIdentityProvider()
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/IamIdentityProvider.cs:8]: of course private -> just changed it.

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/CompanyApplication.cs [hunk 3]
```diff
@@ -1,21 +1,38 @@
...
 using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace CatenaX.NetworkServices.PortalBackend.PortalEntities.Entities
 {
-    public class CompanyApplication : BaseEntity
+    public class CompanyApplication
     {
         public CompanyApplication()
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/CompanyApplication.cs:10]: yes, private -> changed.

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/CompanyIdentityProvider.cs [hunk 4]
```diff
@@ -5,16 +5,18 @@ namespace CatenaX.NetworkServices.PortalBackend.PortalEntities.Entities
     public class CompanyIdentityProvider
     {
         public CompanyIdentityProvider() {}
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/CompanyIdentityProvider.cs:7]: changed

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/Consent.cs [hunk 5]
```diff
@@ -1,35 +1,46 @@
...
+﻿using CatenaX.NetworkServices.PortalBackend.PortalEntities.Enums;
+using System;
 using System.ComponentModel.DataAnnotations;
 
 namespace CatenaX.NetworkServices.PortalBackend.PortalEntities.Entities
 {
-    public class Consent : BaseEntity
+    public class Consent
     {
         public Consent() {}
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Entities/Consent.cs:9]: changed

**Review by wira-mms — APPROVED** [review]

**Pull request merged**